src/ComputeSharp/Graphics/Resources/Abstract/Texture2D{T}.cs

```diff
@@ -223,17 +223,85 @@ internal void CopyTo(ref T destination, int size, int x, int y, int width, int h
         }
     }
 
+    /// <summary>
+    /// Reads the contents of the specified range from the current <see cref="Texture2D{T}"/> instance and writes them into a target <see cref="Texture2D{T}"/> instance.
+    /// </summary>
+    /// <param name="destination">The target <see cref="Texture2D{T}"/> instance to write data to.</param>
+    /// <param name="destinationOffsetX">The horizontal offset within <paramref name="destination"/>.</param>
+    /// <param name="destinationOffsetY">The vertical offset within <paramref name="destination"/>.</param>
+    /// <param name="sourceOffsetX">The horizontal offset in the source texture.</param>
+    /// <param name="sourceOffsetY">The vertical offset in the source texture.</param>
+    /// <param name="width">The width of the memory area to copy.</param>
+    /// <param name="height">The height of the memory area to copy.</param>
+    internal void CopyTo(Texture2D<T> destination, int sourceOffsetX, int sourceOffsetY, int destinationOffsetX, int destinationOffsetY, int width, int height)
+    {
+        GraphicsDevice.ThrowIfDisposed();
+
+        ThrowIfDisposed();
+
+        destination.ThrowIfDeviceMismatch(GraphicsDevice);
+        destination.ThrowIfDisposed();
+
+        Guard.IsInRange(sourceOffsetX, 0, Width, nameof(sourceOffsetX));
+        Guard.IsInRange(sourceOffsetY, 0, Height, nameof(sourceOffsetY));
+        Guard.IsInRange(destinationOffsetX, 0, destination.Width, nameof(destinationOffsetX));
+        Guard.IsInRange(destinationOffsetY, 0, destination.Height, nameof(destinationOffsetY));
+        Guard.IsBetweenOrEqualTo(width, 1, Width, nameof(width));
+        Guard.IsBetweenOrEqualTo(height, 1, Height, nameof(height));
+        Guard.IsBetweenOrEqualTo(width, 1, destination.Width, nameof(width));
+        Guard.IsBetweenOrEqualTo(height, 1, destination.Height, nameof(height));
```

Comment:
Are these needed? Aren't these a subset of the next cases of offset+width/height (as negative offsets aren't allowed)?

Reply:
They're needed, yeah. You could have (in fact, I accidentally had in a test, which is how I spotted this ahah) a valid offset that's some positive number, and then a negative length that's small enough. If you don't check the length explicitly, you could end up with a final offset + length being valid, but with the length still being a negative number 🙂